src/langchain_google_memorystore_redis/vector_store.py

```diff
@@ -0,0 +1,721 @@
+        """
+        command = (
+            f"FT.DROPINDEX {index_name} {'KEEPDOCS' if index_only else ''}".strip()
+        )
+        client.execute_command(command)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        batch_size: int = 1000,
+        **kwargs: Any,
+    ) -> List[str]:
+        """
+        Adds a collection of texts along with their metadata to a vector store,
+        generating unique keys for each entry if not provided.
+
+        Args:
+            texts (Iterable[str]): An iterable collection of text documents to be added to the vector store.
+            metadatas (Optional[List[dict]], optional): An optional list of metadata dictionaries,
+                where each dictionary corresponds to a text document in the same order as the `texts` iterable.
+                Each metadata dictionary should contain key-value pairs representing the metadata attributes
+                for the associated text document.
+            batch_size (int, optional): The number of documents to process in a single batch operation.
+                This parameter helps manage memory and performance when adding a large number of documents.
+                Defaults to 1000.
+            **kwargs (Any): Additional keyword arguments for extended functionality. This includes:
+                - 'keys' or 'ids' (List[str], optional): Custom identifiers for each document. If provided,
+                the length of this list should match the length of `texts`. If not provided, the system
+                will generate unique identifiers.
+
+        Returns:
+            List[str]: A list containing the unique keys or identifiers for each added document. These keys
+                can be used to retrieve or reference the documents within the vector store.
+
+        Note:
+            If both 'keys' (or 'ids') and 'metadatas' are provided, they must be of the same length as the
+            `texts` iterable to ensure each document is correctly associated with its metadata and identifier.
+        """
+        # Generate or extend keys/IDs for the documents
+        keys_or_ids = kwargs.get("keys", kwargs.get("ids", []))
+        # Ensure there's a unique ID for each text document
+        keys_or_ids = (keys_or_ids + [str(uuid.uuid4()) for _ in texts])[
+            len(keys_or_ids) :
+        ]
+        # Fallback for empty metadata
+        metadatas = metadatas if metadatas is not None else [{} for _ in texts]
+        # Generate embeddings for all documents
+        embeddings = self.embedding_service.embed_documents(list(texts))
+
+        ids = []
+        pipeline = self.client.pipeline(transaction=False)
+        for i, bundle in enumerate(
+            zip_longest(keys_or_ids, texts, embeddings, metadatas), start=1
+        ):
+            key, text, embedding, metadata = bundle
+            key = self.key_prefix + key
```

Comment:
Here it seems adding back the prefix? We can generate what we want for keys_or_ids in line 383 at once?

Reply:
My intent is for every key to be prefixed for the purpose of isolation - I am seriously thinking of removing the prefix customization capability next. WDYT?

Reply:
removed key prefixing customization